pulseapi/entries/views.py

```diff
@@ -4,13 +4,54 @@
 
 import django_filters
 from rest_framework import (filters, status)
-from rest_framework.decorators import detail_route
+from rest_framework.decorators import detail_route, api_view
 from rest_framework.generics import ListCreateAPIView, RetrieveAPIView
 from rest_framework.pagination import PageNumberPagination
 from rest_framework.response import Response
 
 from pulseapi.entries.models import Entry
 from pulseapi.entries.serializers import EntrySerializer
+from pulseapi.users.models import UserFavorites
+
+# FIXME: TODO: Remove the handling for GET, so that only PUT works
+@api_view(['GET','PUT'])
+def toggle_favorite(request, entryid):
```

Comment:
Just so I understand what's going on here because I've seen this before, can you explain line 17 and 18 from a django standpoint? Why is this a function and not a view class? Is 17 the indicator that it's a generic GET/PUT view which does nothing special?